### ISSUE TITLE
Menu Editor move classes intended for <a> from <li>

### ISSUE THREAD

**Issue by Vitzkrieg** [wp-bootstrap/wp-bootstrap-navwalker#270]
The WP menu editor allows you to add custom classes to menu items, but some of the bootstrap classes are intended to be placed on `<a>` instead of `<li>` which is where WP adds them. This could be worked around by adding additional CSS in the theme's style to replicate the same styling, but that would not be efficient.

#### Steps to reproduce the issue:
Add classes using the Menu editor in WP

#### What I expected:
Classes such as "btn", "navbar-btn", "btn-primary", etc. to be added on `<a>`

#### What happened instead:
Classes are added to `<li>` which is the default for WP menu editor

#### Solution
I have some working code to move any "btn" classes from `<li>` to `<a>`. This may be better done with an array of all classes to move, but this worked for my issue that I needed fixed. I will create a new branch as well.

```
			$class_names = $value = '';
			$classes = empty($item->classes) ? array() : (array) $item->classes;
			// move any bootstrap classes set in the menu editor intended for <a> from <li>
			$litoa = [];
			foreach ($classes as $index => $value) {
				if (!empty($value) && strpos($value, 'btn') !== FALSE) {
					$litoa[] = $value;
					unset($classes[$index]);
				}
			}
			$classes[] = 'nav-item menu-item-' . $item->ID;
			$class_names = join(' ', apply_filters('nav_menu_css_class', array_filter($classes), $item, $args));
```
```
			// add any classes that were originally on <li> to <a>
			if (count($litoa) > 0) {
				$atts['class'] .= ' ' . implode(' ', $litoa);
			}
			$atts = apply_filters('nav_menu_link_attributes', $atts, $item, $args);
			$attributes = '';
```
*Edited to show tags in code blocks

**Comment by Unit60** [comment]
@Vitzkrieg I'm having the same issue... but not fluid in PHP... I also want to have the class-name (added in the WP customize menu) applied to the 'a' rather than the 'li' tag above... and not sure how to implement your script?!
Any help??

**Comment by Vitzkrieg** [comment]
Some of the code I posted is from the master branch, so you can look through and see where my new code fits in.

**Comment by pattonwebz** [comment]
I've been reworking icon, header, divider etc support and I do have some workable code in the v4 branch that does a similar thing to what is described here. That rework partially because of exploration into this issue. It turns out that proper implementation for a feature like this required a substantial rework because it made a lot of the other code incompatible or redundant.

I tagged this as won't fix for BS3 (it is almost working in the v4 branch) but I will definitely backport it to the BS3 version when I have it working well.

**Comment by proevilz** [comment]
@pattonwebz 

So does this work in the BS4 branch somewhere yet? Currently having the same issue...

**Comment by pattonwebz** [comment]
Hey again @legionio, 

The latest work on this can be found in this PR: https://github.com/wp-bootstrap/wp-bootstrap-navwalker/pull/307

It moves some specific classes from the `<li>` to the inside `<a>`. I do not recall specifically what is moved but I believe I started with `.disabled` and `.dropdown-header` support.

**Comment by pattonwebz** [comment]
I looked into doing this while I reworked icon and link mod support for the v4 branch and it's possible using the method outlined by OP however the list of specific classes that could need moved is pretty large. Too large to bundle an array of them and test explicitly for moving all of them.

I'm choosing to stick with just having icon support and support for the standard bootstrap link mods for now but I will add in a filter for you to test for additional classes sometime in the future.
